metricflow/test/snapshots/test_dataflow_to_sql_plan.py/SqlQueryPlan/DuckDbSqlClient/test_join_to_time_spine_node_with_offset_window__plan0_optimized.sql

```diff
@@ -0,0 +1,43 @@
+-- Join to Time Spine Dataset
+SELECT
+  subq_9.ds AS ds
+  , subq_9.metric_time AS metric_time
+  , subq_9.listing AS listing
+  , booking_value * 0.05 AS booking_fees
+FROM (
+  -- Date Spine
+  SELECT
+    ds AS metric_time
+  FROM ***************************.mf_time_spine subq_11
+  WHERE ds BETWEEN CAST('2020-01-01' AS TIMESTAMP) AND CAST('2021-01-01' AS TIMESTAMP)
+) subq_11
+LEFT OUTER JOIN (
+  -- Compute Metrics via Expressions
+  SELECT
+    ds
+    , metric_time
+    , listing
+    , booking_value * 0.05 AS booking_fees
+  FROM (
+    -- Read Elements From Data Source 'bookings_source'
+    -- Pass Only Elements:
+    --   ['booking_value', 'ds', 'listing']
+    -- Metric Time Dimension 'ds'
+    -- Aggregate Measures
+    SELECT
+      ds
+      , ds AS metric_time
+      , listing_id AS listing
+      , SUM(booking_value) AS booking_value
+    FROM (
+      -- User Defined SQL Query
+      SELECT * FROM ***************************.fct_bookings
+    ) bookings_source_src_10001
+    GROUP BY
+      ds
+      , ds
+      , listing_id
+  ) subq_9
+) subq_10
+ON
+  subq_11.metric_time - INTERVAL 10 day = subq_10.metric_time
```

Comment:
Two things:

1. The aliases on the outer select columns don't look right, they should be coming from subq_10 or subq_11, not subq_9. I think I see the issue in the visitor, will comment there.
2. I don't think this is offsetting the values. What we want to do with the offset is re-assign the metric value to a different date, which I believe means the metric_time has to come from the time_spine with offset applied, not the original metric dataset. Is that right?

So it'd be like:

```
SELECT spine.metric_time, metric.metric_value
FROM (
    SELECT metric_time
    FROM time_spine
    WHERE <constraint>
) spine
LEFT OUTER JOIN (
  SELECT metric_time, metric_value
  FROM metric_data_set
) metric
ON spine.metric_time - INTERVAL 10 day = metric.metric_time
```

Reply:
Ah you're right, should be selecting metric_time from `subq_11` instead of `subq_9`! Will update.

Reply:
Updated if you want to check out the new snapshots.